lib/querystring.js

```diff
@@ -18,7 +18,7 @@ QueryString.unescapeBuffer = function(s, decodeSpaces) {
   var n, m, hexchar;
 
   for (var inIndex = 0, outIndex = 0; inIndex <= s.length; inIndex++) {
-    var c = s.charCodeAt(inIndex);
+    var c = inIndex < s.length ? s.charCodeAt(inIndex) : NaN;
```

Comment:
Would there be drawbacks to changing the `<=` to `<' in the loop test?  You'd have to replicate some of the `out[outIndex++]` assignments after the loop but it would keep the loop body simple.  I guess you can also accomplish that with a `s/NaN/0/.